mirgecom/euler.py

```diff
@@ -80,25 +82,18 @@ def _facial_flux(discr, eos, q_tpair, local=False):
         "all_faces."  If set to *True*, the returned fluxes are not projected to
         "all_faces"; remaining instead on the boundary restriction.
     """
-    dim = discr.dim
-
     actx = q_tpair[0].int.array_context
+    dim = discr.dim
 
-    flux_int = inviscid_flux(discr, eos, q_tpair.int)
-    flux_ext = inviscid_flux(discr, eos, q_tpair.ext)
-
-    # Lax-Friedrichs/Rusanov after [Hesthaven_2008]_, Section 6.6
-    flux_avg = 0.5*(flux_int + flux_ext)
-
+    euler_flux = partial(inviscid_flux, discr, eos)
     lam = actx.np.maximum(
         compute_wavespeed(dim, eos=eos, cv=split_conserved(dim, q_tpair.int)),
         compute_wavespeed(dim, eos=eos, cv=split_conserved(dim, q_tpair.ext))
     )
-
     normal = thaw(actx, discr.normal(q_tpair.dd))
-    flux_weak = (
-        flux_avg @ normal
-        - 0.5 * lam * (q_tpair.ext - q_tpair.int))
+
+    # todo: user-supplied flux routine
```

Comment:
Should we modify the arguments to `_facial_flux` to accept an incoming flux function? Something like
```python
def _facial_flux(discr, eos, q_tpair, numerical_flux_fct, local=False):
    ...
    kwargs = {
        "compute_flux": euler_flux,
        "normal": normal,
        "lam": lam
    }
    flux_weak = numerical_flux_fct(q_tpair, **kwargs)
    ...
```
This is more or less opening up a pathway to drop in other choices of numerical fluxes for the inviscid terms, such as HLLC or Roe.

Reply:
Yes, but not in *this* PR. That change (i.e. the equivalent one) is coming in a later PR.